libcxx/test/libcxx/algorithms/alg.sorting/assert.sort.invalid_comparator.pass.cpp

```diff
@@ -52,204 +51,234 @@
 
 class ComparisonResults {
```

Comment:
We need to take the parts of this test that would reproduce an OOB access inside `std::sort` & friends to a new test file. And that one should be enabled whenever the valid-element-access checks are enabled.

And then this test becomes a test for the debug mode, so we should be able to remove `XFAIL: libcpp-hardening-mode=debug`.